### DISCUSSION TITLE
Some questions about architecture

### DISCUSSION THREAD

**Discussion by Chris202305** [lnis-uofu/OpenFPGA#1339] [Q&A]
The question is about the “L=4” in the k6_N10_40nm.xml followed by "Routing architecture", What is the meaning of "L=4" ? 

<img width="264" alt="1693982393107" src="https://github.com/lnis-uofu/OpenFPGA/assets/142561908/fda49e3e-76d8-47c9-beaa-974908c814af">

**Comment by tangxifan** [comment]
@Chris202305 I suggest to read Prof. Vaughn's [PhD thesis](https://tspace.library.utoronto.ca/handle/1807/12758), which explains many principles and details about FPGA architectures (most of them are still valid nowadays!)

**Comment by tangxifan** [comment]
Also you may find some explantion about wire segments in VTR's documentation 
https://docs.verilogtorouting.org/en/latest/arch/reference/#wire-segments
